beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -487,8 +500,66 @@ impl<T: EthSpec> ValidatorMonitor<T> {
                         u64_to_i64(validator.withdrawable_epoch),
                     );
                 }
+
+                // Add non finalized missed blocks for the monitored validators for the current slot of the current epoch
+                let current_slot = state.slot();
+                self.missed_blocks.contains(&(current_epoch, i as u64, current_slot))
+                    .then(|| {
+                        metrics::inc_counter_vec(
+                            &metrics::VALIDATOR_MONITOR_MISSED_NON_FINALIZED_BLOCKS_TOTAL,
+                            &[current_epoch.to_string().as_str(), current_slot.to_string().as_str(), id],
+                        );
+                    });
+            }
+        }
+    }
+
+    fn add_validators_missed_blocks(&mut self, current_epoch: Epoch, state: &BeaconState<T>) {
+        // Determine missed (non-finalized) blocks (can contain false positives)
+        let range_of_slot = (T::slots_per_epoch() as usize - MISSED_BLOCK_LAG_SLOTS) - 1;
+        for n in 0..range_of_slot {
+            let slot = state.slot() - Slot::new(n as u64);
+            let prev_slot = slot - 1;
```

Comment:
I think for this range we want something like:

- start: `state.slot() - MISSED_BLOCK_LAG_SLOTS - T::slots_per_epoch()`
- end: `state.slot() - MISSED_BLOCK_LAG_SLOTS`

i.e. a range of length `slots_per_epoch` that ends 4 blocks before the state's current slot. At the moment the loop will start at `state.slot()` (when `n=0`), which negates the effect of the buffer -- we want to wait for a few slots to pass before declaring a block missed.